presto-native-execution/presto_cpp/main/types/tests/FunctionMetadataTest.cpp

```diff
@@ -0,0 +1,156 @@
+  EXPECT_EQ(metadataList.size(), numSignatures);
+
+  for (auto i = 0; i < numSignatures; i++) {
+    json j = metadataList.at(i);
+    EXPECT_EQ(j.at("functionKind"), protocol::FunctionKind::AGGREGATE);
+    EXPECT_EQ(j.at("docString"), functionName);
+    EXPECT_EQ(j.at("schema"), "default");
+    auto outputType = j.at("outputType");
+    VELOX_CHECK(
+        ((outputType == "real") || (outputType == "double") ||
+         (outputType == "DECIMAL(38,a_scale)") || (outputType == "bigint")));
+    auto paramTypes = j.at("paramTypes");
+    VELOX_CHECK(
+        ((paramTypes == std::vector<std::string>({"real"})) ||
+         (paramTypes == std::vector<std::string>({"double"})) ||
+         (paramTypes ==
+          std::vector<std::string>({"DECIMAL(a_precision,a_scale)"})) ||
+         (paramTypes == std::vector<std::string>({"tinyint"})) ||
+         (paramTypes == std::vector<std::string>({"smallint"})) ||
+         (paramTypes == std::vector<std::string>({"integer"})) ||
+         (paramTypes == std::vector<std::string>({"bigint"}))));
+
+    auto routineCharacteristics = j.at("routineCharacteristics");
+    EXPECT_EQ(
+        routineCharacteristics.at("language"), protocol::Language({"CPP"}));
+    EXPECT_EQ(
+        routineCharacteristics.at("determinism"),
+        protocol::Determinism::DETERMINISTIC);
+    EXPECT_EQ(
+        routineCharacteristics.at("nullCallClause"),
+        protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+
+    auto aggregateMetadata = j.at("aggregateMetadata");
+    auto intermediateType = aggregateMetadata.at("intermediateType");
+    VELOX_CHECK(
+        (intermediateType == "double") || (intermediateType == "VARBINARY") ||
+        (intermediateType == "bigint"));
+    EXPECT_EQ(aggregateMetadata.at("isOrderSensitive"), false);
+  }
+}
+
+TEST(FunctionMetadataTest, testRank) {
+  window::prestosql::registerAllWindowFunctions("");
+  std::string functionName = "rank";
+  json metadataList = getJsonMetadataForFunction(functionName);
+  const auto numSignatures = 1;
+  EXPECT_EQ(metadataList.size(), numSignatures);
+
+  json j = metadataList.at(0);
+  EXPECT_EQ(j.at("functionKind"), protocol::FunctionKind::WINDOW);
+  EXPECT_EQ(j.at("docString"), functionName);
+  EXPECT_EQ(j.at("schema"), "default");
+  auto outputType = j.at("outputType");
+  VELOX_CHECK(((outputType == "integer") || (outputType == "bigint")));
+  VELOX_CHECK(j.at("paramTypes") == std::vector<std::string>({}));
+
+  auto routineCharacteristics = j.at("routineCharacteristics");
+  EXPECT_EQ(routineCharacteristics.at("language"), protocol::Language({"CPP"}));
+  EXPECT_EQ(
+      routineCharacteristics.at("determinism"),
+      protocol::Determinism::DETERMINISTIC);
+  EXPECT_EQ(
+      routineCharacteristics.at("nullCallClause"),
+      protocol::NullCallClause::CALLED_ON_NULL_INPUT);
+}
+
+TEST(FunctionMetadataTest, testRadians) {
+  functions::prestosql::registerArithmeticFunctions("");
+  std::string functionName = "radians";
```

Comment:
We should test more functions here. Include scalar functions of each type and template type parameters.